internal/util/fetch/fetch.go

```diff
@@ -94,21 +94,62 @@ func (c Command) validate(kf *kptfilev1.KptFile) error {
 	return nil
 }
 
+type Cloner struct {
```

Comment:
Would be good to add some context or background thinking how this new struct will be used and especially any deviation from current workflow.

Reply:
Added some context